arm-asazure/2016-05-06/swagger/asazure.json

```diff
@@ -0,0 +1,536 @@
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": { "$ref": "#/definitions/RpProvisionResult" }
+          },
+          "204": {
+            "description": "No Content",
+            "schema": { "$ref": "#/definitions/RpProvisionResult" }
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "patch": {
+        "description": "Updates the current state of the specified Analysis Services server as specified by the request body",
+        "tags": [ "RpProvision" ],
+        "operationId": "RpProvision_UpdateDetailsAsync",
+        "parameters": [
+          {
+            "name": "analysisServerName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true,
+            "description": "Name of the Analysis Services server"
+          },
+          {
+            "name": "updateRequest",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/RpServerUpdateRequest" },
+            "description": "Request object for updating the server"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": { "$ref": "#/definitions/RpProvisionResult" }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AnalysisServices/servers": {
+      "get": {
+        "description": "Gets all the Analysis Services servers for the given resource group",
+        "tags": [ "RpProvision" ],
+        "operationId": "RpProvision_GetAllServersForResourceGroupAsync",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": { "$ref": "#/definitions/ProvisionResultList" }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AnalysisServices/servers": {
+      "get": {
+        "description": "List all the Analysis Services servers for the given subscription",
+        "tags": [ "RpProvision" ],
+        "operationId": "RpProvision_ListAllServersForSubscriptionAsync",
```

Comment:
@athipp Can you address Amar's comment here? Async is an implementation detail for the client that gets generated. It shouldn't appear in the name of the operation, because it gets appended in languages that support async, for the Async versions of the call:

See this Batch Account_Create operation: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-batch/2015-12-01/swagger/BatchManagement.json#L44

And the resulting client library:
https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Batch/Microsoft.Azure.Management.Batch/Generated/AccountOperationsExtensions.cs#L43
and
https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Batch/Microsoft.Azure.Management.Batch/Generated/AccountOperationsExtensions.cs#L73

In C#, you have an async version and a non-async version. Your async version would be called `RpProvision.ListAllServersForSubscriptionAsyncAsync`, while your non-async call would wrongly be called `RpProvision.ListAllServersForSubscriptionAsync`

Also, please rename this to `RpProvision_ListServers`. This operation provides servers across the subscription and recommended naming pattern is `RpProvision_ListServers` (see here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md)